Arrangement_on_surface_2/include/CGAL/Arr_geodesic_arc_on_sphere_traits_2.h

```diff
@@ -1763,113 +1796,108 @@ class Arr_geodesic_arc_on_sphere_traits_2 : public Kernel_ {
-      const Point_2& l2 = xcv2.left();
-      const Point_2& r2 = xcv2.right();
+      CGAL_precondition((ce != ARR_MIN_END) ||
+                        (xcv1.left().is_mid_boundary() &&
+                         xcv2.left().is_mid_boundary()));
+      CGAL_precondition((ce != ARR_MAX_END) ||
+                        (xcv1.right().is_mid_boundary() &&
+                         xcv2.right().is_mid_boundary()));
 
-      // If xcv1 is vertical, xcv1 coincides with the discontinuity arc:
-      if (xcv1.is_vertical()) {
-        CGAL_precondition(!l1.is_no_boundary());
-        CGAL_precondition(!r1.is_no_boundary());
+      // If the curves lie on the same plane return EQUAL.
+      const Kernel& kernel = m_traits;
+      const Direction_3& n1 = xcv1.normal();
+      const Direction_3& n2 = xcv2.normal();
+      if (xcv1.is_directed_right() == xcv2.is_directed_right()) {
+        if (kernel.equal_3_object()(n1, n2)) return EQUAL;
       }
-
-      // If xcv2 is vertical, xcv2 coincides with the discontinuity arc:
-      if (xcv2.is_vertical()) {
-        CGAL_precondition(!l2.is_no_boundary());
-        CGAL_precondition(!r2.is_no_boundary());
+      else {
+        auto opposite_3 = kernel.construct_opposite_direction_3_object();
+        auto opposite_n2 = opposite_3(n2);
+        if (kernel.equal_3_object()(n1, opposite_n2)) return EQUAL;
       }
 
-      if (ce == ARR_MIN_END) {
-        // Handle the south pole. It has the smallest y coords:
-        if (l1.is_min_boundary())
-          return (l2.is_min_boundary()) ? EQUAL : SMALLER;
-        if (l2.is_min_boundary()) return LARGER;
+      // The queves do not lie on the same plane!
+      const Point_2& l1 = xcv1.left();
+      const Point_2& l2 = xcv2.left();
+      const Point_2& r1 = xcv1.right();
+      const Point_2& r2 = xcv2.right();
 
+      if (ce == ARR_MIN_END) {
         // None of xcv1 and xcv2 endpoints coincide with a pole:
         Comparison_result cr = m_traits.compare_y(l1, l2);
         if (cr != EQUAL) return cr;
 
         // If Both arcs are vertical, they overlap:
-        if (xcv1.is_vertical() && xcv2.is_vertical()) return EQUAL;
         if (xcv1.is_vertical()) return LARGER;
         if (xcv2.is_vertical()) return SMALLER;
 
-        // Non of the arcs is verticel. Thus, non of the endpoints coincide
-        // with a pole.
-        // Compare the y-coord. at the x-coord of the most left right-endpoint.
-        CGAL_assertion(r1.is_no_boundary());
-        CGAL_assertion(r2.is_no_boundary());
-
-        if (m_traits.compare_xy(r1, r2) == LARGER) {
-          // use r2 and xcv1:
-          Oriented_side os =
-            m_traits.oriented_side(xcv1.normal(), r2);
-          return (os == ON_ORIENTED_BOUNDARY) ? EQUAL :
-            (xcv1.is_directed_right()) ?
-            ((os == ON_NEGATIVE_SIDE) ? LARGER : SMALLER) :
-            ((os == ON_NEGATIVE_SIDE) ? SMALLER : LARGER);
+        // There are 4 cases based on the sign of the z component of the normals
```

Comment:
That is a bit special, and I do not think we have that in the kernel. The predicate here seems to be the sign of the x-coordinate of the cross-product, so that is:
```c++
  sign(n1.y() * n2.z() - n1.z() * n1.y())
```
That is indeed a waste of time to compute the two other coordinates of the cross-product.